src/script/api/script_vehiclelist.hpp

```diff
@@ -20,7 +20,32 @@
  */
 class ScriptVehicleList : public ScriptList {
 public:
+#ifdef DOXYGEN_API
 	ScriptVehicleList();
+
+	/**
+	 * Apply a filter when building the list.
+	 * @param filter_function The function which will be doing the filtering.
+	 * @param params The params to give to the filters (minus the first param,
+	 *  which is always the index-value).
+	 * @note You can write your own filters and use them. Just remember that
+	 *  the first parameter should be the index-value, and it should return
+	 *  a bool.
+	 * @note Example:
+	 *  ScriptVehicleList(ScriptVehicle.IsInDepot);
```

Comment:
These are two different examples, right? Maybe explain that in comments.

Reply:
I followed ScriptList::Valuate() documentation.